### ISSUE TITLE
devise token auth + Save Facebook auth_hash info in database

### ISSUE THREAD

**Issue by gustavoguimaraes** [lynndylanhurley/devise_token_auth#326]
I am using Devise auth token gem and ng-token auth for authenticating my single page app. Besides registering with email there is the option to sign-up via Facebook. It works.

My problem is that as for my Fb request scope I set to get the user_friends, email and public_info, however when I get the request I don't see the auth_hash request.env['omniauth.auth']. This is described in the omniauth-facebook gem documentation.

Basically I want to save what FB gives back about the user in my database. How would you do it?

https://stackoverflow.com/questions/31833159/devise-token-auth-save-facebook-info

**Comment by tylerbobella** [comment]
:+1: I am having the same issue

**Comment by gustavoguimaraes** [comment]
According to the [devise wiki about omniauth] (https://github.com/plataformatec/devise/wiki/OmniAuth:-Overview) expects the response to have request.env['omniauth.auth'] which I can't find. 
Instead I have in my response request.env['action_dispatch.request.unsigned_session_cookie'] which does not give me all the info in the auth_hash that [omniauth-facebook provides] (https://github.com/mkdynamic/omniauth-facebook#auth-hash)

**Comment by tylerbobella** [comment]
@gustavoguimaraes I could only find information in the auth_hash but when I change the scope it will not return new data. It seems as though the scope values are not getting passed correctly.

**Comment by gustavoguimaraes** [comment]
@tylerbobella totally. 

Within request.env['action_dispatch.request.unsigned_session_cookie'] 

I see
...
{"provider"=>"facebook",
   "uid"=>"******",
   "info"=>{"name"=>"John Doe", "image"=>"http://graph.facebook.com/99999/picture"},
   .....

but no other info from public_profile or scopes values

**Comment by consti** [comment]
@gustavoguimaraes I'm running into the same problem – I'm especially missing the `email` attribute in the `info` hash. Did you find a solution?

**Comment by consti** [comment]
@gustavoguimaraes I was able to resolve this issue by adding the desired fields to the provider hash:

```ruby
# /config/initializers/omniauth.rb
Rails.application.config.middleware.use OmniAuth::Builder do
  provider(
    :facebook,
    ENV['FACEBOOK_KEY'],
    ENV['FACEBOOK_SECRET'],
    scope: 'email',
    info_fields: 'email,first_name,last_name')
end
```

Now `omniauth-facebook` actually requests (and receives) `email, first_name, last_name` and passes it forward to the `DeviseTokenAuth::OmniauthCallbacksController` (make sure to define your own if you want to use `first_name,last_name` etc.).

**Comment by booleanbetrayal** [comment]
Closing this issue since it's somewhat out of scope for this gem and a configuration example for omniauth-facebook has been provided. Can re-open if need be though.
